src/Symfony/Component/Worker/Loop/Loop.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Worker\Loop;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Worker\Exception\StopException;
+use Symfony\Component\Worker\MessageCollection;
+use Symfony\Component\Worker\Router\RouterInterface;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Loop implements LoopInterface, ConfigurableLoopInterface
+{
+    private $router;
+    private $eventDispatcher;
+    private $logger;
+    private $name;
+    private $options;
+
+    private $stopped;
+    private $startedAt;
+    private $lastHealthCheck;
+
+    public function __construct(RouterInterface $router, EventDispatcherInterface $eventDispatcher = null, LoggerInterface $logger = null, $name = 'unnamed', array $options = array())
+    {
+        $this->router = $router;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->logger = $logger;
+        $this->name = $name;
+        $this->options = array_replace(array(
+            'loop_sleep' => 200000,
+            'health_check_interval' => 10,
+        ), $options);
+
+        $this->stopped = false;
+        $this->startedAt = time();
+        // Fake the date to trigger right now a new health check.
+        $this->lastHealthCheck = 0;
+    }
+
+    public function run()
+    {
+        if (null !== $this->logger) {
+            $this->logger->notice('Worker {worker} started.', array(
```

Comment:
Requiring a null logger make it harder to initialize the class by hand. But I usually do the following in the constructor to avoid the `if (null !== $this->logger)` everywhere:

```php
if (null === $this->logger) {
    $this->logger = new NullLogger();
}
```